### PR TITLE
Contracts Page Shell

### PR DESCRIPTION
**Pull request by nitaliano**
**Description**

This PR starts to implement the contracts page. It adds all of the components needed outside of the app section which will be done in a follow on PR.

This is an overview of what the page looks like right now
Screenshot 2024-03-20 at 2 52 26 PM

Variant of what it looks like if the progress bar has any progress
Screenshot 2024-03-20 at 2 54 43 PM

### CODE REVIEW

File: apps/dapp-console/app/settings/components/ClaimedRebates.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  txHash: Hash
+  value: bigint
+}
+
+export type CliamedRebateProps = {
+  rebates?: Rebate[]
+}
+
+export const ClaimedRebates = ({
+  rebates = [
```
Comment (nitaliano) [inline apps/dapp-console/app/settings/components/ClaimedRebates.tsx:30]: mocked the object for now until we hookup the backend

File: apps/dapp-console/app/settings/components/CoinbaseVerificationBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import Image from 'next/image'
+
+import { Text } from '@eth-optimism/ui-components/src/components/ui/text'
+
+export const CoinbaseVerificationBanner = () => (
+  <div className="flex flex-row bg-[#d6e4ff] w-full p-8 rounded-xl items-center">
```
Comment (imranjami) [inline apps/dapp-console/app/settings/components/CoinbaseVerificationBanner.tsx:6]: the static `bg-[#d6e4ff]` will look a little off in dark mode, instead we can use `bg-blue-500/20` to set the opacity of the token to 20% which achieves the same visual on light mode + gives us a nice dark mode variant as well.

File: apps/dapp-console/app/settings/components/CoinbaseVerificationBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      src="/logos/coinbase-logo.png"
+      width={40}
+      height={40}
+      alt="Coinbase Logo"
+    />
+    <div className="flex flex-col">
+      <Text as="p" className="text-base text-[#3374DB] font-semibold">
+        Coinbase Verification is required for rebates
+      </Text>
+      <Text as="p" className="text-sm text-[#3374DB]">
```
Comment (imranjami) [inline apps/dapp-console/app/settings/components/CoinbaseVerificationBanner.tsx:18]: Same here as above

File: apps/dapp-console/app/settings/contracts/page.tsx [hunk 4]
```diff
@@ -1,3 +1,36 @@
...
+  return (
+    <div className="flex flex-col gap-2">
+      <Text className="text-lg font-semibold">Your Apps</Text>
+
+      <div className="flex flex-col gap-6">
+        <Apps />
+
+        <Button
+          variant="outline"
+          className="flex justify-between w-[120px] mb-8 rounded"
```
[the remark this answers is not in the document]
Comment (nitaliano) [reply apps/dapp-console/app/settings/contracts/page.tsx:22]: yea i have it in a few places will remove it

**Review by imranjami — APPROVED** [review]

File: apps/dapp-console/app/settings/components/CoinbaseVerificationBanner.tsx (hunk 2 above)
Comment (nitaliano) [reply apps/dapp-console/app/settings/components/CoinbaseVerificationBanner.tsx]: sick i didn't realize you can handle opacity this way, the changes look nice 👍 thanks!

**Pull request merged**